functions/operatingHours.ts

```diff
@@ -21,25 +21,85 @@ enum DaysOfTheWeek {
   Sunday = 7,
 }
 
-type OperatingInfo = {
+type OperatingInfoEntry = {
   timezone: string; // the timezone the helpline uses
   holidays: { [date: string]: string }; // a date (in MM/DD/YYYY format) - holiday name object to specify which days are holidays for the helpline
   operatingHours: { [channel: string]: { [day in DaysOfTheWeek]: OperatingShift[] } }; // object that pairs numbers representing weekdays to open and close shifts
 };
 
+// The root contains OperatingInfoEntry info (default, support legacy) plus an "offices" entry, which maps OperatingInfoEntry to a particular office
+type OperatingInfo = OperatingInfoEntry & {
+  offices: {
+    [office: string]: OperatingInfoEntry;
+  };
+};
+
 type EnvVars = {
   OPERATING_INFO_KEY: string;
 };
 
 export type Body = {
   channel?: string;
+  office?: string;
 };
 
 const isOpen =
   (timeOfDay: number) =>
   (shift: OperatingShift): boolean =>
     timeOfDay >= shift.open && timeOfDay < shift.close;
 
+const getStatusFromEntry = (operatingInfoEntry: OperatingInfoEntry, channel: string) => {
+  // TODO: message
+  if (!operatingInfoEntry || !operatingInfoEntry.operatingHours[channel]) {
+    throw new Error(
+      `Operating Info not found for channel ${channel}. Check OPERATING_INFO_KEY env vars and a matching OperatingInfo json file for it.`,
+    );
+  }
+
+  const { timezone, holidays, operatingHours } = operatingInfoEntry;
+
+  const timeOfDay = parseInt(
+    moment().tz(timezone).format('Hmm'), // e.g 123 for 1hs 23m, 1345 for 13hs 45m
+    10,
+  );
+  const dayOfWeek = moment().tz(timezone).isoWeekday() as DaysOfTheWeek;
+  const currentDate = moment().tz(timezone).format('MM/DD/YYYY');
+
+  if (currentDate in holidays) {
+    return 'holiday';
+  }
+
+  const isInOpenShift = isOpen(timeOfDay);
+  const isOpenNow = operatingHours[channel][dayOfWeek].some(isInOpenShift);
+
+  if (isOpenNow) {
+    return 'open';
+  }
+
+  return 'closed';
+};
+
+const getOperatingStatus = (operatingInfo: OperatingInfo, channel: string, office?: string) => {
+  const { offices, ...operatingInfoRoot } = operatingInfo;
+
+  // TODO: should we loudly error if office is provided but there's no operatingInfo.offices entry?
```

Comment:
I would say falling back to the default is fine in either of these cases, it will probably make updates easier to coordinate if things don't blow up when there's a transient mismatch (e.g. you've updated the asset but not the flow yet or vice versa)